cloud/aws/events/events.go

```diff
@@ -0,0 +1,137 @@
+package events
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ref "k8s.io/client-go/tools/reference"
+	clusterapiclientsetscheme "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/scheme"
+)
+
+// ObjectRec holds a contained event recorder
+type ObjectRec struct {
+	rec *record.EventRecorder
+	obj runtime.Object
+}
+
+type key int
+
+const (
+	contextKey key = 0
+
+	// Normal represents a normal event
+	Normal = v1.EventTypeNormal
+
+	// Warning event
+	Warning = v1.EventTypeWarning
+
+	// Failure event
+	Failure = "Failure"
+
+	// Pending event
+	Pending = "Pending"
+)
+
+var (
+	globalEventRec *record.EventRecorder
```

Comment:
I'm ok if this is a DefaultEventRecorder, similar to how http does DefaultHttpClient, but we should have a way to make a new one and not use mutable globals wherever possible.

What about if the API were like this:

```go
recorder := events.DefaultObjectRecorder
// or
recorder, err := events.NewRecorder(param1, param2)
// check error
objectRecorder, err := events.NewStdObjRecorder(recorder)
// check error
```

Reply:
Hah, yeah. Felt dirty writing that.